Makefile

```diff
@@ -261,6 +268,19 @@ ifeq ($(ENABLE_LIBYOSYS),1)
 TARGETS += libyosys.so
 endif
 
+ifeq ($(ENABLE_PYOSYS),1)
+	ifeq ($(PYTHON_MAJOR_VERSION),3)
+		LDLIBS += -lpython$(PYTHON_VERSION)m -lboost_python-py$(subst .,,$(PYTHON_VERSION)) -lboost_system -lboost_filesystem
```

Comment:
Arch Linux calls the Boost Python library boost_python37 rather than boost_python-py37. There is significant inter-distro variation here, see all the different cases that we ended up having to detect in nextpnr: https://github.com/YosysHQ/nextpnr/blob/master/CMakeLists.txt#L132-L186 - I'm not sure if there is a better way than bruteforcing all the possibilities here.

Reply:
Thanks for the heads-up. I implemented 4 versions now:

1. boost_python-py<version>
2. boost_python-py<major_version>
3. boost_python<version>
4. boost_python<major_version>

If none of those work, the user is prompted to define BOOST_PYTHON_LIB manually.
cc270ea81bc7209e9e1673e6c3dbd5db6d3ceee8

Reply:
Thanks, this works well for me now. One small comment is the auto-detection prints
```
/usr/bin/ld: cannot find -lboost_python-py37
collect2: error: ld returned 1 exit status
/usr/bin/ld: cannot find -lboost_python-py3
collect2: error: ld returned 1 exit status
/usr/bin/ld: cannot find -lboost_python-py37
collect2: error: ld returned 1 exit status
/usr/bin/ld: cannot find -lboost_python-py3
collect2: error: ld returned 1 exit status
```
at the moment, which could be confusing as it looks like an error. Perhaps you should redirect the output to /dev/null or similar?

Reply:
Suppressed error output in e19981ab6111765baa5b2ab7a16c92278130fd8b